### DISCUSSION TITLE
Followed trouble shooting advice, and still receiving 404 error when trying to deploy to GitHub Pages

### DISCUSSION THREAD

**Discussion by lukeqanderson** [community/community#23538] [Other Feature Feedback, Questions, & Ideas]
<p>I have been trying to publish my portfolio website to GitHub pages under the following link:</p>
<p><a href="https://lukeqanderson.github.io/portfolio/" class="onebox" target="_blank" rel="noopener nofollow ugc">https://lukeqanderson.github.io/portfolio/</a></p>
<p>However, every time I try to deploy it shows that it failed. I have changed my main repository to master where all my files are located, ensured that master is selected under the pages setting and deleted the README file to test if that resolves the issue.</p>
<p>Here is the link to my repository:</p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/lukeqanderson/portfolio">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/lukeqanderson/portfolio" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/5714058/181077791-84c007b7-48a0-4994-9210-97e0627ac8bf.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c0e243f715d463a377120da17d14d8cef6055b4f_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c0e243f715d463a377120da17d14d8cef6055b4f_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/c/0/c0e243f715d463a377120da17d14d8cef6055b4f.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/c/0/c0e243f715d463a377120da17d14d8cef6055b4f_2_10x10.png"></div>

<h3><a href="https://github.com/lukeqanderson/portfolio" target="_blank" rel="noopener nofollow ugc">GitHub - lukeqanderson/portfolio: A portfolio website created with HTML, CSS...</a></h3>

  <p>A portfolio website created with HTML, CSS and JavaScript to showcase responsive functionality without implementing any additional styling frameworks.  - GitHub - lukeqanderson/portfolio: A portfol...</p>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>Any help would be greatly appreciated!</p>
<p>Thanks,<br>
Luke.</p>

**Comment by jfullstackdev** [comment]
<p>I just checked your page and it is still 404</p>
<p>create a new repo like <code>test</code> then make a simple <code>README.md</code>, deploy the site at root, then let’s see,</p>
<p>by this way, before you put any build files you know that your GitHub Pages is working, then you can push all of your build files after</p>

**Comment by lukeqanderson** [comment, marked as answer]
<p>Was able to resolve this issue. For those running into similar issues, I had received the following error:</p>
<p>Error: Error: No uploaded artifact was found!<br>
Error: Error: No uploaded artifact was found!</p>
<p>With some further investigation, I realized that I had made a submodule accidentally (I’m still learning git), which contained a copy of my repository. Once I deleted the submodule (name “portfolio” in my case), I was able to deploy with no issues.</p>

**Comment by methodphoto** [comment]
<p>A noob with Github Pages here, I too am getting the same errors. What does “I had made a submodule accidentally” mean. I don’t think I’ve done anything like that</p>

**Comment by lukeqanderson** [comment]
<p>Hi methodphoto,</p>
<p>More or less I created a copy of my entire repository within my repository so it was showing up as a submodule. I would push from this submodule to my GitHub. When it came time to try and deploy the site, GitHub wouldn’t deploy it because it wasn’t sure what to do with the submodule. Since I didn’t need it, I deleted it and resolved the problem.</p>
<p>It looks as though your site is now up and deployed. Is it another site that is affected or were you able to resolve the issue?</p>

**Comment by methodphoto** [comment]
<p>Hi Luke</p>
<p>Yes in the end I took the approach of starting again with a minimal site, that worked, and just added new parts bit by bit.<br>
I think because I had a local jekyll running, files were being pushed that interfered with the GitHub pages build.</p>
<p>Thanks for responding  :grinning_face_with_smiling_eyes:</p>

**Comment by prashant-shahi** [comment]
<p>I was experiencing similar issues.</p>
<p>I resolved it by creating <code>gh-page</code> branch and deleting the submodule in that branch while using <code>docs</code> folder for static HTML files.</p>
<p>GitHub staff should look into this newly introduced problem.</p>

**Comment by gulinan** [comment]
<p>Hi all,</p>
<p>While I have some experience on deploying at GitHub pages, this time it also gives me the same error message:</p>
<p><code> </code> <code> [Error: Error: No uploaded artifact was found!]</code> <code> </code></p>
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/gulinan/montpellier_ml_talk2022/runs/4730755929?check_suite_focus=true">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/gulinan/montpellier_ml_talk2022/runs/4730755929?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/56047198/181077795-fb7d2c53-c5ee-451b-a9b2-e44377bca56a.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94cf0a113027e72da26fb2663be709db2700f7d2_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94cf0a113027e72da26fb2663be709db2700f7d2_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/9/4/94cf0a113027e72da26fb2663be709db2700f7d2.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/9/4/94cf0a113027e72da26fb2663be709db2700f7d2_2_10x10.png"></div>

<h3><a href="https://github.com/gulinan/montpellier_ml_talk2022/runs/4730755929?check_suite_focus=true" target="_blank" rel="noopener nofollow ugc">update · gulinan/montpellier_ml_talk2022@488169f</a></h3>

  <p>Contribute to gulinan/montpellier_ml_talk2022 development by creating an account on GitHub.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>for this repo: <a href="https://github.com/gulinan/montpellier_ml_talk2022" class="inline-onebox" rel="noopener nofollow ugc">GitHub - gulinan/montpellier_ml_talk2022</a>.</p>
<p>I will also appreciate any help/suggestion for getting rid of this error (i checked submodule thing, deleted  all the files, created new repo, but nothing solved the problem).</p>
<p>Thank you very much.</p>
<p>Gul</p>

**Comment by lukeqanderson** [comment]
<p>Hi gulinan,</p>
<p>Sorry for the delayed response. I took a look through your repository and came across the following error under jobs → build:</p>
<p>“Liquid Exception: Liquid syntax error (line 113): Variable ‘{{i}’ was not properly terminated with regexp: /}}/ in index.Rmd”</p>
<p>I’m personally not familiar with liquid, but I was able to find a similar issue to the one you are experiencing that looks to be closed:</p>
<aside class="onebox githubissue" data-onebox-src="https://github.com/jekyll/jekyll/issues/5458">
  <header class="source">

      <a href="https://github.com/jekyll/jekyll/issues/5458" target="_blank" rel="noopener nofollow ugc">github.com/jekyll/jekyll</a>
  </header>

  <article class="onebox-body">
    <div class="github-row">
  <div class="github-icon-container" title="Issue">
	  <svg width="60" height="60" class="github-icon" viewbox="0 0 14 16" aria-hidden="true"><path d="M7 2.3c3.14 0 5.7 2.56 5.7 5.7s-2.56 5.7-5.7 5.7A5.71 5.71 0 0 1 1.3 8c0-3.14 2.56-5.7 5.7-5.7zM7 1C3.14 1 0 4.14 0 8s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7zm1 3H6v5h2V4zm0 6H6v2h2v-2z"></path></svg>
  </div>

  <div class="github-info-container">
    <h4>
      <a href="https://github.com/jekyll/jekyll/issues/5458" target="_blank" rel="noopener nofollow ugc">Liquid Exception: Liquid syntax error (line 123): Variable '{{0,1}' was not properly terminated with regexp: /\}\}/ in xxx.md</a>
    </h4>

    <div class="github-info">
      <div class="date">
        opened <span class="discourse-local-date" data-format="ll" data-date="2016-10-06" data-time="19:10:34" data-timezone="UTC">07:10PM - 06 Oct 16 UTC</span>
      </div>

        <div class="date">
          closed <span class="discourse-local-date" data-format="ll" data-date="2016-10-06" data-time="19:26:39" data-timezone="UTC">07:26PM - 06 Oct 16 UTC</span>
        </div>

      <div class="user">
        <a href="https://github.com/livc" target="_blank" rel="noopener nofollow ugc">
          <img alt="livc" src="https://user-images.githubusercontent.com/5714058/181077798-0ac044f0-904f-4670-a8ca-ab95741ff2f5.jpeg" class="onebox-avatar-inline" width="20" height="20">
          livc
        </a>
      </div>
    </div>

    <div class="labels">
        <span style="display:inline-block;margin-top:2px;background-color: #B8B8B8;padding: 2px;border-radius: 4px;color: #fff;margin-left: 3px;">
          frozen-due-to-age
        </span>
    </div>
  </div>
</div>

  <div class="github-row">
    <p class="github-body-container">&lt;!--
  Hi! Thanks for considering to file a bug with Jekyll. Please take the tim<span class="show-more-container"><a href="" rel="noopener" class="show-more">…</a></span><span class="excerpt hidden">e to
  answer the basic questions. You can convert `[ ]` into `[x]` to check boxes (or submit
  and check.) If there is no need for certain fields like output and redirection, please delete
  those headers before submitting. We know not all tickets require those steps.
  Otherwise, please try to be as detailed as possible.

  If you are unsure this is a bug in Jekyll, or this is a bug caused
  by a plugin that isn't directly related to Jekyll, or if this is just
  a generic usage question, please consider asking your question at
  https://talk.jekyllrb.com where non-bug questions go.

  Please make sure to mention an affinity team whose responsibilities
  most closely align with your issue.

  Thanks!
--&gt;
- [x] I believe this to be a bug, not a question about using Jekyll.
- [ ] I updated to the latest Jekyll (or) if on GitHub Pages to the latest `github-pages`
- [ ] I read the CONTRIBUTION file at https://jekyllrb.com/docs/contributing/
- [ ] This is a feature request.

---
- [x] I am on (or have tested on) **_macOS**_ 10+
- [ ] I am on (or have tested on) **_Debian/Ubuntu**_ GNU/Linux
- [ ] I am on (or have tested on) **_Fedora**_ GNU/Linux
- [ ] I am on (or have tested on) **_Arch**_ GNU/Linux
- [ ] I am on (or have tested on) **_Other**_ GNU/Linux
- [ ] I am on (or have tested on) **_Windows**_ 10+

&lt;!--
  Other GNU/Linux includes Scientific GNU/Linux, CentOS GNU/Linux, and others.
  If you are on a minor sub-distro (such as ElementaryOS which does not diverge from
  Ubuntu much, please check the parent distro. Kubuntu, Edubuntu, Lubuntu should
  also be flagged as Ubuntu as their packages come from upstream Ubuntu.
--&gt;

---
- [ ] I was trying to install.
- [ ] There is a broken Plugin API.
- [ ] I had an error on GitHub Pages, and I have reproduced it locally.
- [ ] I had an error on GitHub Pages, and GitHub Support said it was a Jekyll Bug.
- [ ] I had an error on GitHub Pages and I did not test it locally.
- [x] I was trying to build.
- [ ] It was another bug.
## My Reproduction Steps

After`jekyll server`, I get:

```
➜  livc.github.io git:(master) ✗ jekyll server
Configuration file: /Users/zhao/repo/livc.github.io/_config.yml
            Source: /Users/zhao/repo/livc.github.io
       Destination: /Users/zhao/repo/livc.github.io/_site
 Incremental build: disabled. Enable with --incremental
      Generating...
  Liquid Exception: Liquid syntax error (line 123): Variable '{{0,1}' was not properly terminated with regexp: /\}\}/ in /Users/zhao/repo/livc.github.io/_posts/2016-02-07-neu1682.md
jekyll 3.2.1 | Error:  Liquid syntax error (line 123): Variable '{{0,1}' was not properly terminated with regexp: /\}\}/
```

But nothing in line 123 as http://pastebin.com/vZszwbvt

When I change line 134 from
`int dis[4][2]={{0,1},{0,-1},{1,0},{-1,0}};`
to
`int dis[4][2]={ {0,1},{0,-1},{1,0},{-1,0}};`

It fixed !

&lt;!--
  If this error occured on GitHub Pages, please try to provide us with logs,
  and look at them yourself, to determine if this is an actual Jekyll bug. In
  the event you are unsure, file a ticket, however, when you do please provide
  the logs (strip them of personal information.)

  If you have trouble finding your logs, please email support@github.com and
  they will happily help you. If you cannot find logs, please try your best to
  replicate it locally because we cannot fix a problem if we do not know
  exactly what caused it, or within a relatively close distance.
--&gt;

&lt;!--
  Insert the steps you took to for this problem to exist. Such as the
  directories you created and, the full command you ran, and include any
  plugins you have installed, this is very important.

  If your steps are complicated, you can also submit a GitHub
  repository (please no zips, they will be removed and rejected by maintainers,)
  and just supply a command for us to reproduce it ourselves.
--&gt;
## The Output I Wanted

&lt;!--
  Insert the output from the command. Alter it as little as you can.
  The minimum should be personal information. Though we normally don't log
  anything like that so there should be no need to alter it.
--&gt;

/cc include any Jekyll affinity teams here (see https://teams.jekyllrb.com/ for more info)</span></p>
  </div>

  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>The solution seems to be:</p>
<p>" Wrap that whole line with <code>{% raw %}</code> … <code>{% endraw %}</code> to let Liquid know not to try to parse it :+1: "</p>
<p>for your variable “i” on line 113 in index.rmd.</p>
<p>I’m still a GitHub noob, so I apologize if this is something you have already tested. Hope it helps!</p>
<p>Luke.</p>

**Comment by gulinan** [comment]
<p>Hi lukeqanderson, I would like to apologize for replying your email very lately.  This was an extra ordinary week, where i was dealing with paper grading along with the problems which current situation of pandemic brought.</p>
<p>I just tried your solution, but, I could not let it worked. I am also not a guru on this. I will leave it as it is. I would like to thank you very much for your kind reply one more time, and would like to apologize for being late million times. I would appreciate your understanding. Best wishes.</p>

**Comment by HowieWork** [comment]
<p>Hi Luke,<br>
I am having the same issue. I deleted my original repository on GitHub and deployed all over again. But the error still exists. Would you take a look? Thank you so much. <a href="https://github.com/HowieWork/howiework/runs/5011827298?check_suite_focus=true" rel="noopener nofollow ugc">Error link</a></p>

**Comment by airtower-luna** [comment]
<p>I’m afraid you’re looking at the wrong error. The deploy jobs failed simply because the build job failed. The build job seems to have issues with an invalid date format: <a href="https://github.com/HowieWork/howiework/runs/5011820775?check_suite_focus=true#step:4:17" class="inline-onebox">Updates · HowieWork/howiework@a66435d · GitHub</a></p>
<p>Your problem seems unrelated to the original post, in the future please create a new thread for that.</p>

**Comment by HowieWork** [comment]
<p>Thank you so much. Yes, I found the problems.</p>
